sklego/linear.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from typing import Union
+
+import numpy as np
+import scipy.spatial.distance as distance
+from sklearn.base import BaseEstimator, RegressorMixin
+from sklearn.linear_model import LinearRegression
+from sklearn.neighbors import NearestNeighbors
+from sklearn.utils import check_X_y
+from sklearn.utils import check_array
+from sklearn.utils.validation import FLOAT_DTYPES
+from sklearn.utils.validation import check_is_fitted
+
+
+class LoessRegressor(BaseEstimator, RegressorMixin):
+    """
+    Lo(w)ess regressor.
+
+    """
+
+    def __init__(self, weighting_method: Union[str, None] = None, span: float = .1):
+        super().__init__()
+
+        self._check_init_inputs(weighting_method, span)
+
+        self.weighting_method = weighting_method
+        self.span = span
```

Comment:
might be good to have something like `self.allowed_weighting_methods` to confirm which methods are allowed.